shared/general/container-ssh.md

```diff
@@ -1,46 +1,83 @@
+![SSH Terminal](/img/common/device/terminal.png)
 
 If your device is online, select a target and click the blue *>_ Start Terminal session* button. A terminal session should be initiated for you in a second or two. If you would like a bigger window for the terminal, you can click the *Expand* button in the upper-right corner.
 
-__Note:__ To copy and paste in the terminal window,  you cannot use the normal Ctrl + C and Ctrl + V shortcuts. You can either select Copy and Paste from a menu, or use Ctrl + Insert for copy and Shift + Insert for Paste. For MacOS users, ⌘ + C and ⌘ + V work as expected.
+__Note:__ To copy and paste in the terminal window, you cannot use the normal Ctrl + C and Ctrl + V shortcuts. You can either select Copy and Paste from a menu, or use Ctrl + Insert for copy and Shift + Insert for Paste. For MacOS users, ⌘ + C and ⌘ + V work as expected.
 
 ## Using `{{ $names.company.short }} ssh` from the CLI
 
+__Note:__ Host OS access via the CLI requires {{ $names.cli }} version 6.12.0 or above.
+
 If you prefer to work from the command line, you can use [`{{ $names.company.short }} ssh`][balena-ssh] to connect to your application containers and the host OS. First, you will need to install the [{{ $names.company.lower }} Command Line Interface (CLI)](/tools/cli/). Once that is set up, run the following in your development machine's terminal:
 
 ```shell
 $ {{ $names.company.short }} ssh <device-uuid>
 ```
 
-`<device-uuid>` is the unique identifier for the device you want to access, which can be found on the dashboard.
+`<device-uuid>` is the unique identifier for the device you want to access, which can be found via the dashboard or in the output of the `{{ $names.company.short }} devices` CLI command. By default, SSH access is routed into the host OS shell. However, you can SSH into a service by specifying its name as part of the command:
+
+```shell
+$ {{ $names.company.short }} ssh <device-uuid> main
+```
+
+This also works in multicontainer applications, simply pass the name of the appropriate service as defined in docker-compose.yml you'd like to access the shell for.
+
+For devices running development images on your local network, you can create a direct SSH connection by specifying the hostname using the `<device-uuid>.local` syntax or local IP address of the device. Development images have passwordless root SSH access enabled by default, and so they should never be directly exposed to the public internet. Using {{ $names.company.short }} ssh in this way does not use the balena VPN and instead makes a direct SSH connection to the device. For example:
```

Comment:
> For devices running development images on your local network, you can create a direct SSH connection

I found this a bit confusing, because a "direct SSH connection" (meaning a connection that does not rely on the balenaCloud VPN) can be used with both development images and production images, and it is also not limited to local IP addresses -- a globally routable IP address would work too, although uncommon. I suggest (just a suggestion) reordering and slightly rewording this paragraph and the following two, as follows:

---
> ...
> When an application name or device UUID is used as above, `{{ $names.company.short }} ssh` uses the {{ $names.company.short }} VPN to create a secure tunnel to the device and then forward SSH traffic between the device and your development machine. For this method, you must have a SSH key configured on your development machine and [added to the {{ $names.cloud.lower }} dashboard][add-ssh-key].
>
> If an IP address or a `.local` hostname is used (instead of an application name or device UUID), `{{ $names.company.short }} ssh` establishes a direct connection that does not rely on the {{ $names.company.short }} VPN:
>
```shell
$ balena ssh 192.168.1.23
$ balena ssh <device-uuid>.local
```
>
> This should work without further configuration in the case of devices running a [development balenaOS image](https://www.balena.io/docs/reference/OS/overview/2.x/#dev-vs-prod-images), which allows passwordless root SSH access (and for this reason, should never be directly exposed to the public internet). In the case of a production balenaOS image, an SSH key needs to be present in the `config.json` file of the device. More details about configuring SSH keys in `config.json` may be found [here][config-json-ssh].

Above I have also replaced "development image" with "development balenaOS image", and made it a link.

Reply:
Thank you, this is excellent feedback and I have updated this.